packages/flutter/lib/src/rendering/sliver_group.dart

```diff
@@ -0,0 +1,146 @@
+import 'sliver.dart';
+
+/// A sliver that places multiple sliver children in a linear array along the cross
+/// axis.
+///
+/// Since the extent of the viewport in the cross axis direction is finite,
+/// each sliver is forced to divide up the extent per specification via
+/// [ParentDataWidget] [SliverConstrainedCrossAxis] and [SliverExpanded].
+///
+/// The algorithm for dividing up the cross axis extent is as follows.
+/// Every widget either has constrained cross axis extent or a flex value.
+/// First allocate space to all the widgets with constrained cross axis extent.
+/// Then divide up the remaining space proportional to the widget's flex value.
+/// Flex values can be specified via the [SliverExpanded] widgets, but if
+/// a sliver without [SliverExpanded] or [SliverConstrainedCrossAxis] is passed,
+/// it is assumed to have a flex value of 1, which by itself expands
+/// to take up all of the cross axis extent.
+class RenderSliverCrossAxisGroup extends RenderSliver with ContainerRenderObjectMixin<RenderSliver, SliverPhysicalContainerParentData> {
+  @override
+  void setupParentData(RenderObject child) {
+    if (child.parentData is! SliverPhysicalContainerParentData) {
+      child.parentData = SliverPhysicalContainerParentData();
+    }
+  }
+
+  @override
+  double childMainAxisPosition(RenderSliver child) => 0.0;
+
+  @override
+  double childCrossAxisPosition(RenderSliver child) {
+    switch (constraints.axisDirection) {
+      case AxisDirection.up:
+      case AxisDirection.down:
+        return (child.parentData! as SliverPhysicalContainerParentData).paintOffset.dx;
+      case AxisDirection.left:
+      case AxisDirection.right:
+        return (child.parentData! as SliverPhysicalContainerParentData).paintOffset.dy;
+    }
+  }
+
+  @override
+  void performLayout() {
+    // Iterate through each sliver.
+    // Get the parent's dimensions.
+    final double crossAxisExtent = constraints.crossAxisExtent;
+    assert(crossAxisExtent.isFinite);
+
+    // First, layout each child with flex == 0 or null.
+    int totalFlex = 0;
+    double usedUpExtent = 0.0;
+    RenderSliver? child = firstChild;
+    while (child != null) {
+      final SliverPhysicalContainerParentData childParentData = child.parentData! as SliverPhysicalContainerParentData;
+      final int flex = childParentData.crossAxisFlex ?? 0;
+      if (flex == 0) {
+        // If flex is 0 or null, then the child sliver must provide their own crossAxisExtent.
+        child.layout(constraints, parentUsesSize: true);
+        assert(child.geometry!.crossAxisExtent != null);
+        usedUpExtent += child.geometry!.crossAxisExtent!;
+      } else {
+        totalFlex += flex;
+      }
+      child = childAfter(child);
+    }
+    final double extentPerFlexValue = (crossAxisExtent - usedUpExtent) / totalFlex;
+
+    child = firstChild;
+    double offset = 0.0;
+
+    // At this point, all slivers with constrained cross axis should already be laid out.
```

Comment:
What happens if `usedUpExtent` exceed the available space?

Reply:
I think the behavior should be the same as when a child within a `SizedBox` overflows. Although I'm not too sure how to replicate the behavior. Some code pointers would be helpful.

Reply:
An assertion would be a good first addition for this case.

Since you are using the box Flex as a a model for this, see the paint method of its RenderObject, RenderFlex. It handles painting an overflow indicator on the screen and providing error messages to the user.

Reply:
I've added assertions, let me know what you think of them.